#include <exception>

class PyOFException : std::exception
{
	public :

// =====================================================================
// =====================================================================

		PyOFException ();


// =====================================================================
// =====================================================================


		PyOFException (std::string* input);


// =====================================================================
// =====================================================================


		PyOFException (char* input);


// =====================================================================
// =====================================================================


		PyOFException (const char* input);


// =====================================================================
// =====================================================================


		~PyOFException () throw ();


// =====================================================================
// =====================================================================


		const char* what() const throw();


// =====================================================================
// =====================================================================


	private :
		std::string* m_Message;
};


// =====================================================================
// =====================================================================


void TranslatePyOFException (const PyOFException& e);
